docs/clean/user_guide/executing_a_clean.rst

```diff
@@ -42,28 +42,92 @@ Passing images through CLI
 .. topic:: Providing the images through CLI arguments
 
     * Can provide a new image at every run without modifying the Clean YAML file.
-    * Can provide the image to a specific device or all devices with a specific platform.
+    * Can provide the image to a specific device, to all devices in a group, to all devices with a given OS, or all devices with a specific platform.
+    * `--clean-device-image`, `--clean-os-image`, `--clean-group-image` and `--clean-platform-image` can all be specified simultaneously. Conflicts are resolved by `device > group > platform > os`.
+    * Images specified with the CLI always override images specified in the YAML file (the previous order only applies to images passed on the CLI, e.g. an image passed with `--clean-os-image` will override an image specified under a device block in a Clean YAML file if the OS of the device matches)
```

Comment:
addition to this explanation, might be better to have like `CLI > YAML` similar to CLI one.